frontend/src/app/core/setup/globals/global-listeners/hover-card-trigger.service.ts

```diff
@@ -154,13 +164,50 @@ export class HoverCardTriggerService {
       this.mouseInModal = false;
     }
 
-    if (!this.mouseInModal) {
+    // It is important to check if we are currently showing a hover card. If we closed the modal service without
+    // doing so, we might accidentally close another modal (e.g. share dialog).
+    if (this.isShowingHoverCard && !this.mouseInModal) {
       this.opModalService.close();
+      this.isShowingHoverCard = false;
       // Allow opening this target once more, since it has been orderly closed
       this.previousTarget = null;
     }
   }
 
+  /*
+   * Some elements do not carry the hover data arguments on themselves, but rely on nearby elements to provide them.
+   * For example in the auto completer.
+   *
+   * This method will return the element that actually carries the hover card data.
+   */
+  private getDataSource(el:HTMLElement) {
+    if (el.className.includes('op-hover-card--data-on-sibling-principal')) {
+      const candidate = this.findSiblingElementWithHoverCardData(el);
+      if (candidate) {
+        return candidate as HTMLElement;
+      }
+    }
+
+    return el;
+  }
+
+  private findSiblingElementWithHoverCardData(el:HTMLElement) {
+    const sibling = jQuery(el).siblings('.op-principal').get(0);
+    if (sibling) {
+      return this.childElementWithHoverCardTrigger(sibling);
+    }
+
+    return null;
+  }
+
+  private childElementWithHoverCardTrigger(principal:Element) {
+    return principal.querySelector('.op-hover-card--preview-trigger');
+  }
+
+  private parseHoverCardUrl(el:HTMLElement) {
+    return el.getAttribute('data-hover-card-url');
+  }
```

Comment:
You rely heavily on a certain structure for this to work, but that is not clear when you just see the class. So whenever somewhen changes the structure in the autocompleter and e.g wrap the principal in a div, that will not work any more without the person noticing it.

I don't really get why all of this is necessary. You wrote that you could not add the hoverCard information directly on the element but why not?

Reply:
We had a call about this and Henriette suggested a great way to overcome the obstacles I found while trying to place hovercard data directly onto the span. With these suggestions, all of the complicated methods above could be removed 🎉